sdk/python/kfp/compiler/main.py

```diff
@@ -116,6 +116,12 @@ def main():
   if args.py:
     compile_pyfile(args.py, args.function, args.output, not args.disable_type_check)
   else:
+    import warnings
```

Comment:
My gut feeling is that maybe a more manageable way to do this is by annotating compile_package() as deprecated. See
https://pypi.org/project/Deprecated/

It can also allow us to provide some other meta info. It's up to you.